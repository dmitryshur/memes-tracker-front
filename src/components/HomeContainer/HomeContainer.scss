.home-container {
  display: flex;
  flex-direction: row;
  height: 100vh;
  overflow: hidden;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  transition: transform 1s;

  &.newest-opened {
    transform: translateX(84%);
  }

  &.liked-opened {
    transform: translateX(-84%);
  }

  .sliding-bg {
    animation: move-bg 120s linear infinite alternate;
    background-image: url('/images/home-bg.jpeg');
    background-repeat: repeat-x;
    background-size: contain;
    height: 100%;
    left: 0;
    position: absolute;
    top: 0;
    width: 9000px;
    z-index: 0;
  }
}

@keyframes move-bg {
  0% {
    transform: translate3d(0, 0, 0);
  }
  100% {
    transform: translate3d(-3000px, 0, 0);
  }
}
