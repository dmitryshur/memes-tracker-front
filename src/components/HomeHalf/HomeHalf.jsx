import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { resetSections } from "../../actions/visuals/visuals";
import "./HomeHalf.scss";
import HalfTitles from "../HalfTitles/HalfTitles";
import HalfContent from "../HalfContent/HalfContent";

export const HomeHalf = props => {
  const chosenState = props.chosen ? "opened" : "closed";

  const handleArrowClick = () => {
    props.resetSections();
  };

  return (
    <div className={`home-half home-half-${props.section} ${chosenState}`}>
      <div className={`home-overlay home-overlay-${props.section}`} />
      {props.section === "newest" && <HalfTitles chosen={props.chosen} />}
      <div
        className={`back-arrow back-arrow-${props.section}`}
        onClick={handleArrowClick}
        role="button"
        onKeyPress={handleArrowClick}
        tabIndex="0"
      />
      <HalfContent section={props.section} openState={chosenState} />
    </div>
  );
};

HomeHalf.propTypes = {
  section: PropTypes.string.isRequired,
  chosen: PropTypes.bool,
  resetSections: PropTypes.func
};

HomeHalf.defaultProps = {
  chosen: false,
  resetSections: () => {}
};

const mapDispatchToProps = dispatch => ({
  resetSections: () => dispatch(resetSections())
});

export default connect(undefined, mapDispatchToProps)(HomeHalf);
