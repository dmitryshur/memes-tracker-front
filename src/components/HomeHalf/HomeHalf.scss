$newest-bg: #47a3da;
$liked-bg: #ffffff;

.home-half {
  height: 100%;
  opacity: 0.95;
  width: 50%;
  z-index: 1;
  position: relative;

  .home-overlay {
    height: 100%;
    position: absolute;
    width: 100%;
    z-index: 0;

    &-newest {
      background-color: $newest-bg;
    }

    &-liked {
      background-color: $liked-bg;
    }
  }

  .back-arrow {
    background-size: cover;
    height: 40px;
    opacity: 0;
    outline: none;
    pointer-events: none;
    position: absolute;
    top: 40px;
    transform: translateX(-50%);
    transition: opacity 0.5s;
    width: 40px;

    &-newest {
      background-image: url('/images/arrow-right.svg');
      left: 20%;
    }

    &-liked {
      background-image: url('/images/arrow-left.svg');
      right: calc(20% - 40px);
    }
  }

  &.opened {
    .back-arrow {
      opacity: 0.7;
      pointer-events: all;

      &:hover {
        cursor: pointer;
        opacity: 1;
      }
    }
  }

}
