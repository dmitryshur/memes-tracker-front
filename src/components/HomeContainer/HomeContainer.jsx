import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import "./HomeContainer.scss";
import { getChosenSection } from "../../selectors/visuals/visuals";
import HomeHalf from "../HomeHalf/HomeHalf.jsx";

export const HomeContainer = props => (
  <div
    className={
        props.chosenSection === ""
          ? "home-container"
          : `home-container ${props.chosenSection}-opened`
      }
  >
    <div className="sliding-bg" />
    <HomeHalf section="newest" chosen={props.chosenSection === "newest"} />
    <HomeHalf section="liked" chosen={props.chosenSection === "liked"} />
  </div>
  );

const mapStateToProps = state => ({
  chosenSection: getChosenSection(state)
});

HomeContainer.propTypes = {
  chosenSection: PropTypes.string
};

HomeContainer.defaultProps = {
  chosenSection: ""
};

export default connect(mapStateToProps)(HomeContainer);
