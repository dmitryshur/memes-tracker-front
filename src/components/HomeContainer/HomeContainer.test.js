import React from 'react';
import { shallow } from 'enzyme';
import { HomeContainer } from './HomeContainer';
import { HomeHalf } from '../HomeHalf/HomeHalf';

test('testing HomeContainer component', () => {
  const wrapper = shallow(<HomeContainer />);
  expect(wrapper).toMatchSnapshot();
});

test('testing HomeContainer component with newest section as the chosen section', () => {
  const wrapper = shallow(<HomeContainer chosenSection="newest" />);
  expect(wrapper.hasClass('newest-opened')).toBe(true);
  expect(wrapper).toMatchSnapshot();
});

test('testing HomeContainer component with liked section as the chosen section', () => {
  const wrapper = shallow(<HomeContainer chosenSection="liked" />);
  expect(wrapper.hasClass('liked-opened')).toBe(true);
  expect(wrapper).toMatchSnapshot();
});
