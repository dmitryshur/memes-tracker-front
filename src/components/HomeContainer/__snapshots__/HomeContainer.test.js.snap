// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing HomeContainer component 1`] = `
<div
  className="home-container"
>
  <div
    className="sliding-bg"
  />
  <Connect(HomeHalf)
    chosen={false}
    section="newest"
  />
  <Connect(HomeHalf)
    chosen={false}
    section="liked"
  />
  <Connect(NavBar)
    navbarStatus="closed"
  />
</div>
`;

exports[`testing HomeContainer component with liked section as the chosen section 1`] = `
<div
  className="home-container liked-opened"
>
  <div
    className="sliding-bg"
  />
  <Connect(HomeHalf)
    chosen={false}
    section="newest"
  />
  <Connect(HomeHalf)
    chosen={true}
    section="liked"
  />
  <Connect(NavBar)
    navbarStatus="closed"
  />
</div>
`;

exports[`testing HomeContainer component with newest section as the chosen section 1`] = `
<div
  className="home-container newest-opened"
>
  <div
    className="sliding-bg"
  />
  <Connect(HomeHalf)
    chosen={true}
    section="newest"
  />
  <Connect(HomeHalf)
    chosen={false}
    section="liked"
  />
  <Connect(NavBar)
    navbarStatus="closed"
  />
</div>
`;
